### PR TITLE
fix sample editing

### PR DESCRIPTION
**Pull request by stufisher**
Closes https://jira.diamond.ac.uk/browse/SCI-10049

### CODE REVIEW

**Review by NigelWilson — APPROVED** [review]

**Comment by John-Holt-Tessella** [comment]
Also closes LIMS-78

File: api/src/Page/Sample.php [hunk 1]
```diff
@@ -1928,8 +1925,11 @@ function _update_sample()
             $dcps = $this->db->pq("SELECT dataCollectionPlanId FROM BLSample_has_DataCollectionPlan
                                 WHERE blSampleId = :1", array($this->arg('sid')));
 
-            if (sizeof($dcps)) {
-                foreach ($dcps as $dcp) {
+            if(sizeof($dcps)) {
+                $maxLocation = $this->_get_current_max_dcp_plan_order($this->args['CONTAINERID']);
```
Comment (ndg63276) [inline api/src/Page/Sample.php:1929]:
```suggestion
```
Is this line needed?

File: api/src/Page/Sample.php [hunk 2]
```diff
@@ -1928,8 +1925,11 @@ function _update_sample()
...
+                $maxLocation = array_key_exists('CONTAINERID', $this->args) ? $this->_get_current_max_dcp_plan_order($this->args['CONTAINERID']) : array(0);
+
```
Comment (ndg63276) [inline api/src/Page/Sample.php:1931]:
```suggestion
                $maxLocation = sizeof($maxLocation) ? $maxLocation : -1;
```
Don't we need this line?

File: api/src/Page/Sample.php (hunk 1 above)
Comment (John-Holt-Tessella) [reply api/src/Page/Sample.php]: No but the line `$maxLocation = $maxLocation ? $maxLocation + 1 : 0;` but I didn't like it. So instead I have moved all the code for determining if there is alread an plan order into that function and called it get next index.
Could you review and I will put it on ipsyb-dev-5

**Review by ndg63276 — APPROVED** [review]
Works for me

**Pull request merged**